packages/@webex/common/src/browser-detection.js

```diff
@@ -27,11 +26,12 @@ const mockDetectionObject = {
 };
 
 const createDetectionObject = (results) => {
-  const getOSName = () => results?.getOSName() ?? '';
-  const getOSVersion = () => results?.getOSVersion() ?? '';
+  const getOSName = () => results?.getOSName() ?? browserDetection.unableToGetOSName;
+  const getOSVersion = () => results?.getOSVersion() ?? browserDetection.unableToGetOSVersion;
 
-  const getBrowserName = () => results?.getBrowserName() ?? '';
-  const getBrowserVersion = () => results?.getBrowserVersion() ?? '';
+  const getBrowserName = () => results?.getBrowserName() ?? browserDetection.unableToGetBrowserName;
+  const getBrowserVersion = () =>
+    results?.getBrowserVersion() ?? browserDetection.unableToGetOSVersion;
```

Comment:
Should this be `defaults` and not `error messages`? i.e. maybe the `getOSName()` method should return `unknown` when it is not available?

@chrisadubois , thoughts?

Reply:
I'm also curious if the `results` object here goes stale (collected by the garbage collector) and begins returning `null`. This could be a potential root cause for this issue, but just a thought.

Reply:
agreed on unknown vs an error message